Update module

PrestaSafe opened the pull request:
Fix translations, update logo, change displayName

mickaelandrieu commented:
Hello @PrestaSafe,

some comments here:

* there is no need to put PHPDoc if autogenerated: sadly this brings no value except noise;
* care about the variable names, it's really hard to figure out what is done here: I've updated your contribution, hope this helps you :+1:
* I'v asked an additionnal QA validation, if @BilalDr is OK I'm gonna merge it: good job.

Mickaël

marionf commented:
Hello @PrestaSafe 

I have some issues with your PR as you can see below

![capture d ecran_517](https://user-images.githubusercontent.com/13449658/47646311-d66d2800-db73-11e8-98e9-2de7a1a684d2.png)

![capture d ecran_519](https://user-images.githubusercontent.com/13449658/47646318-da994580-db73-11e8-8b40-314ea9249e10.png)

![capture d ecran_520](https://user-images.githubusercontent.com/13449658/47646323-dec56300-db73-11e8-8e12-e45f4f3a472d.png)

PrestaSafe commented:
@marionf did you try to reset the module ?

marionf commented:
Yes + empty cache and private navigation

PrestaSafe commented:
ok i found why it didn't works, now it should be ok for you @marionf But, i don't have issue on the token error.

Thanks.

marionf commented:
@PrestaSafe 

I still have the issue with the default menu if I change my language

![capture d ecran_526](https://user-images.githubusercontent.com/13449658/47661842-219a3180-db9a-11e8-9632-8021b1da8d07.png)

![capture d ecran_527](https://user-images.githubusercontent.com/13449658/47661856-26f77c00-db9a-11e8-9274-c7f9bd624c22.png)

Also, the issue #22 isn't fixed by this PR

![capture d ecran_528](https://user-images.githubusercontent.com/13449658/47662326-fb28c600-db9a-11e8-8f50-e3bdd6a72f2b.png)

![capture d ecran_529](https://user-images.githubusercontent.com/13449658/47662334-ff54e380-db9a-11e8-9929-b1225d4e9b44.png)

![capture d ecran_530](https://user-images.githubusercontent.com/13449658/47662345-04199780-db9b-11e8-87f5-ffb4b608d99a.png)

BilalDr commented on translations/fr.php:
```diff
@@ -4,8 +4,8 @@
-$_MODULE['<{seoprestamenu}prestashop>seoprestamenu_893454ce3bd8a49290d9a8be4a934b15'] = 'SeoPresta - Menu';
-$_MODULE['<{seoprestamenu}prestashop>seoprestamenu_9fd4add4397a872b26e5152250207efc'] = 'SeoPrestaMenu - un module de menu 100% seo friendly';
+$_MODULE['<{seoprestamenu}prestashop>seoprestamenu_ba93cdcf5c33b46aaa90dfd31d5fc4f8'] = 'Smart Menu';
+$_MODULE['<{seoprestamenu}prestashop>seoprestamenu_1830d22bcd2cab4fc177ef8022c90075'] = 'Smart Menu - nouveau module de menu pour PrestaShop !';
```
The description should be more clear. Can you write something like  "Ajoute un menu seo friendly sur votre site ecommerce"

BilalDr commented on seoprestamenu.php:
```diff
@@ -39,7 +39,7 @@ public function __construct()
         $this->tab = 'administration';
         $this->version = '1.0.0';
-        $this->author = 'seopresta';
+        $this->author = 'SeoPresta.com';
```
Can you write "SeoPresta" instead of SeoPresta.com  ?
 Because we shouldn't have a website in this section.

PrestaSafe commented:
@marionf i've fixed it, it's really strange but i found why, i've somethings have it and sometimes not.
But for PR #22 it's ok for me, see the sreenshots. 
![capture d ecran 2018-10-30 a 10 34 05](https://user-images.githubusercontent.com/4788787/47709031-bb5fee00-dc2f-11e8-8a7e-2b01d98aad5f.png)
![capture d ecran 2018-10-30 a 10 34 37](https://user-images.githubusercontent.com/4788787/47709033-bbf88480-dc2f-11e8-82cc-c90e5536497e.png)
![capture d ecran 2018-10-30 a 10 34 33](https://user-images.githubusercontent.com/4788787/47709034-bbf88480-dc2f-11e8-9f8b-1a4435edf121.png)
![capture d ecran 2018-10-30 a 10 34 22](https://user-images.githubusercontent.com/4788787/47709035-bbf88480-dc2f-11e8-8763-7fa592437c0d.png)
![capture d ecran 2018-10-30 a 10 34 16](https://user-images.githubusercontent.com/4788787/47709036-bbf88480-dc2f-11e8-8a2e-d20ec61381f9.png)
![capture d ecran 2018-10-30 a 10 34 09](https://user-images.githubusercontent.com/4788787/47709038-bbf88480-dc2f-11e8-8410-34620ffe34aa.png)
@BilalDr i've changed author name. 
@mickaelandrieu for phpdoc, i thought it was a good idea, and good practise no ?

marionf commented:
Issue #22 is if you change a product name, category name or page name already in the menu, the name doesn't change in BO and FO

https://drive.google.com/file/d/13Byse4Y_ZFPEf7_vp4GwRwjgAb1lqVj1/view?usp=sharing

PrestaSafe commented:
ok i see, like i said, this feature need to change all of menu engine. We will make it in a futur update. Thanks

mickaelandrieu commented:
> What's wrong with autogenerated PHPDoc? Since PHP doesn't have static typing this can be very helpful.

>  for phpdoc, i thought it was a good idea, and good practise no ?

I really don't see the value of such PHPDocs I've removed from your code:

```
/**
        * Undocumented function
        *
        * @param [type] $node
        * @param integer $level
        * @param [type] $array
        * @return void
        */
        public function constructTreeNode($node, $level = 0, &$array)
```
I'm ok with PHPDocs if they provide information! Something like:

```
      /**
        * Create a tree node from node used in ...
        *
        * @param string $node
        * @param integer $level
        * @param array $array
        * @return void
        */
        public function constructTreeNode($node, $level = 0, &$array)
```
But from my experience, I've noticed even when people do efforts creating meaningful PHPDocs, the PHPDocs become wrong once people change or adapt the function without updating PHPDocs at the same time: so the PHPDocs mislead people on the use of functions.

In an ideal world where PrestaShop have PHP 7.1 minimal requirements, we could enforce strong type hints almost everywhere and PHPDocs could be only used to give some hints:

 ```
public function constructTreeNode(string $node, int $level = 0, array &$array) : void
```

With right naming on functions and good development practices (a function does one thing and one thing well, we could almost remove any PHPDocs, don't you think?

firstred commented:
> With right naming on functions and good development practices (a function does one thing and one thing well, we could almost remove any PHPDocs, don't you think?

I don't agree. Using the right naming on functions does not make up for the lack of static typing. It is not a good idea for an e-commerce app such as PrestaShop to keep using dynamic typing, because most of the time we have to interact with strongly typed APIs. Type juggling is hell if you can't adjust your local code, I can tell you that from experience!

Like you said, from PHP 7.1 on you can declare proper strict typing, so for a smooth transition it would be to useful to start using the PHPDocs already. Most IDEs understand them and it is actually standardized. I don't know what the upper example is supposed to be, but those are definitely not valid PHPDoc tags.

mickaelandrieu commented:
>  Using the right naming on functions does not make up for the lack of static typing.

I was not clear: I want better naming in addition to strong typing 👍

firstred commented:
Great, let's do that! :smile:
